### PR TITLE
Add warning rule for Git links

### PR DESCRIPTION
**Pull request by apinnick**
Closes https://github.com/redhat-documentation/vale-at-red-hat/issues/828

### CODE REVIEW

**Comment by apinnick** [comment]
@aireilly I think I need your help. I can't seem to get the validation rules to work.

**Comment by aireilly** [comment]
I think this needs its own rule. What about: 

```yml
---
extends: existence
message: Do not include a link to '%s' in documentation source unless it is explicitly approved.
link: https://redhat-documentation.github.io/vale-at-red-hat/docs/main/reference-guide/github-links/
ignorecase: true
level: error
scope: raw
action:
  name: remove
tokens:
    - 'https:\/\/gitlab\.com\/.*'
    - 'https:\/\/github\.com\/.*'
```

Note Red Hat rules also need their own small reference page: https://redhat-documentation.github.io/vale-at-red-hat/docs/main/reference-guide/reference-guide/

**Comment by apinnick** [comment]
> I think this needs its own rule. What about:
> 
...

I can give it a try.

**Comment by aireilly** [comment]
So your rules are looking for a slug after the URL, there will fire the rule: 
```
https://github.com/project
https://gitlab.com/project
```
these will not:
```
https://github.com
https://gitlab.com
```

**Comment by apinnick** [comment]
> So your rules are looking for a slug after the URL, there will fire the rule:

It worked! Will this rule really work?

**Comment by apinnick** [comment]
> Are we sure that this directive (No Git links) applies to all projects under the Red Hat umbrella?

It's in the [RHSSG](https://redhat-documentation.github.io/supplementary-style-guide/#external-links).

**Comment by apinnick** [comment]
Shouldn't I be able to see this new rule in the surge.sh preview?

**Comment by aireilly** [comment]
No you need to create a page + update the nav.adoc. See https://github.com/redhat-documentation/vale-at-red-hat/pull/634 for reference

File: .vale/fixtures/RedHat/GitLinks/.vale.ini [hunk 1]
```diff
@@ -0,0 +1,5 @@
+; Vale configuration file to test the `GitLinks` rule
+StylesPath = ../../../styles
+MinAlertLevel = error
```
Comment (aireilly) [inline .vale/fixtures/RedHat/GitLinks/.vale.ini:3]: Can we make this `MinAlertLevel = warning` Anyone that implements Vale in the CI will get failing builds if we merge this and they have errors. We had strong push back in the past about adding new Error rules.

Comment (apinnick) [reply .vale/fixtures/RedHat/GitLinks/.vale.ini:3]: Will do.

File: .vale/styles/RedHat/GitLinks.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+extends: existence
+message: Do not include a link to '%s' in documentation source unless it is explicitly approved.
+link: https://redhat-documentation.github.io/vale-at-red-hat/docs/main/reference-guide/gitlinks/
+ignorecase: true
+level: error
```
[the remark this answers is not in the document]
Comment (aireilly) [reply .vale/styles/RedHat/GitLinks.yml:6]: If a project wants to set this as an error they can update the project vale.ini

**Review by aireilly — APPROVED** [review]
/lgtm

**Pull request merged**